packages/ui-kit/src/components/Counter/Counter.tsx

```diff
@@ -89,13 +98,20 @@ export const CounterComponent = (props: CounterProps) => {
     if (!isLoadingStatic) {
       handlePropsMismatch()
     }
-  }, [isStatic, value, query, isLoadingStatic])
+  }, [isStatic, value, query, isLoadingStatic, accessTokenFromProvider, isLoadingAccessToken])
 
-  if (error || propsMismatch) {
+  React.useEffect(() => {
+    if (isAccessTokenError) {
+      onExpiredToken?.()
+    }
+  }, [isAccessTokenError, onExpiredToken])
```

Comment:
This part looks repetitive. What if we encapsulate this logic in `AccessTokenProvider` itself?

For instance, we can define a new state to store token error and pass it as a param to useAccessToken, e.g.
```
const [isAccessTokenError, setIsAccessTokenError] = React.useState(false)
const { accessToken: accessTokenFromProvider, loading, error } = useAccessToken({ isAccessTokenError })
```

And just trigger `onExpiredToken` directly in `AccessTokenProvider` when `isAccessTokenError` is true.

Reply:
It will also allow merging `fetch` and `onExpiredToken` methods in `AccessTokenProvider` 🤔